python/BioSimSpace/Protocol/_AToM.py

```diff
@@ -0,0 +1,3287 @@
+                A prepared AToM system.
+        data : dict
+            The AToM data dictionary.
+
+        core_alignment : bool
+            Whether to use rigid core restraints to align the two ligands.
+
+        CMCM_restraint : bool
+            Whether to use a center of mass distance restraint.
+
+        restraint : str, [int]
+            The type of restraint to perform. This should be one of the
+            following options:
+                "backbone"
+                     Protein backbone atoms. The matching is done by a name
+                     template, so is unreliable on conversion between
+                     molecular file formats.
+                "heavy"
+                     All non-hydrogen atoms that aren't part of water
+                     molecules or free ions.
+                "all"
+                     All atoms that aren't part of water molecules or free
+                     ions.
+            Alternatively, the user can pass a list of atom indices for
+            more fine-grained control. If None, then no restraints are used.
+
+        force_constant : :class:`GeneralUnit <BioSimSpace.Types._GeneralUnit>`, float
+            The force constant for the restraint potential. If a 'float' is
+            passed, then default units of 'kcal_per_mol / angstrom**2' will
+            be used.
+
+        pos_rest_width : :class:`Length <BioSimSpace.Types.Length>`, float
+                The width of the flat-bottom potential used for coordinate restraint in Angstroms.
+
+        pos_restrained_atoms : [int]
+            The atoms to be restrained.
+
+        align_kf_sep : int, float, str, :class:`GeneralUnit <BioSimSpace.Types._GeneralUnit>`
+            The force constant for the distance portion of the alignment restraint (kcal/(mol A^2)).
+
+        align_k_theta : int, float, str, :class:`Energy <BioSimSpace.Types.Energy>`
+            The force constant for the angular portion of the alignment restaint (kcal/mol).
+
+        align_k_psi : int, float, str, :class:`Energy <BioSimSpace.Types.Energy>`
+            The force constant for the dihedral portion of the alignment restraint (kcal/mol).
+
+        SC_umax : int, float, str, :class:`Energy <BioSimSpace.Types.Energy>`
+            The Umax value for the ATM softcore potential (kcal/mol).
+
+        SC_u0 : int, float, str, :class:`Energy <BioSimSpace.Types.Energy>`
+            The uh value for the ATM softcore potential (kcal/mol).
+
+        SC_a : int, float, str, :class:`Energy <BioSimSpace.Types.Energy>`
+            The a value for the ATM softcore potential.
+
+        cm_kf : int, float, str, :class:`GeneralUnit <BioSimSpace.Types._GeneralUnit>`
+            The force constant for the center of mass distance restraint (kcal/mol/A^2).
+
+        cm_tol : int, float, str, :class:`Length <BioSimSpace.Types.Length>`
+            The tolerance for the center of mass distance restraint (A)."""
```

Comment:
I'm guessing these might map one-to-one with the original Gallicchio implementation, but it might be nice to make the meaning of the kwargs a little clearer and the formatting standardised. For example, I'd prefer if everything was lower snake_case. Things like `core_alignment` seem self explanatory, whereas `CMCM_restraint` might be clearer as `com_distance_restraint` (along with related `com_force_constant` and `com_tolerance`). There are also mixtures of formatting, e.g. `rest`, `restraint`, `restrained`. These could al be `restraint` for consistency.

The latter kwargs are a little confusing if you aren't familiar with the method. (Specifically the `SC` related terms.) I guess these probably will take the default values in most instances, so the user won't really be bothered by them. I'd still prefer if the names and description were a little more informative, though. Perhaps you could add a `Notes` section to the start of the constructor docstring to link to the AToM expressions, or write them if formatting allows? (This might be easier in website API docs.)

There are few inconsistencies in the kwargs, such as both `k` and `kf` being used for force constant, as well as `force_constant` by itself. I'd prefer the same througout, with `force_constant` making things clear. `SC` could be `soft_core` for clarity.

There are a few minor typos too, e.g. `restaint` instead of `restraint`.

Reply:
Kwargs have now been updated to (hopefully) make things clearer. The changes are as follows:
- `CMCM_restraint` -> `com_distance_restraint`
- `cm_kf` -> `com_k`
- `cm_tol` -> `com_restraint_width`
- `align_kf_sep` -> `align_k_distance`
- `pos_rest_width` -> `positional_restraint_width`
- `SC_umax` -> `soft_core_umax`
- `SC_u0` -> `soft_core_u0`
- `SC_a` -> `soft_core_a`
The constants have also been grouped to make it clearer which goes with which (e.g. `com_k` now directly follows `com_distance_restraint` in the arguments)